src/sparse/merkle_tree.rs

```diff
@@ -20,6 +26,21 @@ where
         }
     }
 
+    pub fn load(
+        storage: &'storage mut dyn Storage<Bytes32, Buffer, Error = StorageError>,
+        root: &Bytes32,
+    ) -> Result<Self, Box<dyn std::error::Error>> {
```

Comment:
Why are you boxing the error instead of just using the error directly?

Reply:
it's a way to make the error type generic, we could consider using anyhow::Error for this instead.

Reply:
Something for a future PR then. @bvrooman could you file an issue to use `anyhow` if there isn't one already?

Reply:
Created issue https://github.com/FuelLabs/fuel-merkle/issues/84.

Because the `load` function, as well as other functions in the MT interfaces, act on the underlying storage, they are liable to throw both storage-related errors and MT-related errors. Thus we need to treat the error polymorphically.

The `anyhow` library appears to provide some nice syntax sugar for simplifying the interface.